languages/go/exercises/concept/maps/.docs/introduction.md

```diff
@@ -0,0 +1,58 @@
+In go, `map` is a built-in data type that represent a key/value store or hash table. In other programming language, you might know `map` as `dict` or associative array. If you're not familiar with such concept, map you can think `map` like a dictionary, which every word is the `key` and the definition is the `element` of `map`.
+
+> Before we begin, I'd like to point you to [go spec for map][gospec-map] and [go blog for map][goblog-map] to dig further into `map`.
```

Comment:
The introduction should preferably not link to external documentation. As such, could you remove this line please?

```suggestion
```

Reply:
Oh that's right, but I need 1 link to refer the student to the `comparable` type, I think listing out the what the comparable types are would be too much for this exercise

Reply:
> I need 1 link to refer the student to the comparable type

Does the student need to know this to solve the exercise? I think they don't, as the map types used in the exercise is pre-defined.

Reply:
Partially, they only need to know `string` and `int` type for the keyElement and valueElement. Basically what the docs trying to say is you can use any type for keyElement and valueElement, but for keyElement you can't use `Slice`, `map`, and `function` but the `slice` concept hasn't been introduced yet.

A better approach I think to list what types are available for the keyElement and valueElement, but that might be outdated sometime in the future (who knows?) but that mean I have to put in some keyword that hasn't been introduced, is that alright? I think that's alright, right?

Reply:
@ErikSchierboom: As far as I understood we can have this in the after.md with a link. The `introduction.md` is basically a summary of the necessary parts of the `after.md` to solve the exercise. Is that correct?

If so, @chocopowwwa you can have this in the after.md but just remove it completely from the introduction.md. It is not necessary to know which types can be used as keys in a `map` to solve this, but it is very good to know it in general. All the `after.md` files of completed exercises will be available to the student to re-read later on, to go back to if questions arise. So it is perfectly valid to have this in the after.md.

Note: on my latest exercise I have done (https://github.com/exercism/v3/pull/1438) I have written `after.md` first with everything in there and then just copied it over into introductions.md and removed everything not absolutely necessary. That helped me keep it clean and hopefully not having to change things in 2 documents all the time. (If change requests come up, maybe I should have waited even longer to create the introduction... we'll see).